modules/transitive/src/closure/IdRefVisitor.scala

```diff
@@ -43,11 +44,17 @@ final class IdRefVisitor(
           .getElements()
```

Comment:
not really related to the PR but should `mapShape` be using `visitSeqShape`?

Reply:
not only that but we should probably call it for `visitSeqShape(shape.getKey())` ?

Reply:
that's never going to match `asArrayNode` though, right?

Reply:
I'm like 90% sure this should be `asObjectNode` for map shapes - both for the key and value. I'm not sure how to reproduce it though.

Reply:
Let me have a look, I'll do a final check while I'm there and merge afterward

Reply:
```
=======
shape: (map: `sauce#myTraitMap`)
asArrayNode: Optional.empty
asObjectNode: Optional[software.amazon.smithy.model.node.ObjectNode@b59fc9ec]
asBooleanNode: Optional.empty
asNullNode: Optional.empty
asStringNode: Optional.empty
asNumberNode: Optional.empty
```

Reply:
so you're right